code/game/objects/items/weapons/rpd.dm

```diff
@@ -0,0 +1,303 @@
+	list("key1" = "Supply pipes", "key2" = "2"),
+	list("key1" = "Scrubber pipes", "key2" = "3"),
+	list("key1" = "Devices", "key2" = "4"),
+	list("key1" = "HE pipes", "key2" = "5"))
+	return data
+
+/obj/item/weapon/rpd/Topic(href, href_list, nowindow, state)
+	if(href_list["iconrotation"])
+		iconrotation = text2num(href_list["iconrotation"])
+	else if(href_list["whatpipe"])
+		whatpipe = text2num(href_list["whatpipe"])
+	else if(href_list["whatdpipe"])
+		whatdpipe = text2num(href_list["whatdpipe"])
+	else if(href_list["pipetype"])
+		pipetype = text2num(href_list["pipetype"])
+	else if(href_list["mode"])
+		mode = text2num(href_list["mode"])
+	else if(href_list["refresh"])
+		to_chat(usr, "<span class = 'notice'>You press the reset button, refreshing all settings on the [src].</span>")
+		iconrotation = 0
+		mode = 1
+		pipemenu = 1
+		pipetype = 1
+		whatpipe = 0
+		whatdpipe = 0
+		cooldown = 0
+		Initialiseicons()
+	else
+		return
+	nanomanager.update_uis(src)
+	return
+
+//What the RPD actually does
+
+/obj/item/weapon/rpd/afterattack(atom/target, mob/user as mob, proximity)
+	var/turf/T = get_turf(target)
+	if(!proximity || cooldown == 1)
+		return
+	else if(!isturf(target) && !istype(target, /obj/item/pipe) && !istype(target, /obj/item/pipe_meter) && !istype(target, /obj/item/pipe_gsensor) && !istype(target, /obj/structure/disposalconstruct) || istype(target, /turf/simulated/shuttle))
+		return
+	else if(mode == 1) //Atmos mode
+		if(istype(T, /turf/simulated/wall)) //Drilling into walls takes time and we don't want to do anything else while drilling
+			cooldown = 1
+			playsound(get_turf(src), "sound/weapons/circsawhit.ogg", 50, 1)
+			usr.visible_message("<span class = 'notice'>[usr] starts drilling a hole in [T]...</span>", "<span class = 'notice'>You start drilling a hole in [T]...</span>", "<span class = 'warning'>You hear a drill.</span>")
+			if(do_after(usr, walldelay, target = T))
+				usr.visible_message("<span class = 'notice'>[usr] finishes drilling a hole in [T]!</span>", "<span class = 'notice'>You finish drilling a hole in [T]!</span>", "<span class = 'warning'>You hear clanking.</span>")
+			else
+				cooldown = 0
+				return
+		if(whatpipe == 98)//need to check if it's a meter or a gas sensor separately ayylmao
+			new /obj/item/pipe_gsensor(T)
+			to_chat(usr, "<span class = 'notice'>[src] dispenses a sensor!</span>")
+		else if(whatpipe == 99)
+			new /obj/item/pipe_meter(T)
+			to_chat(usr, "<span class = 'notice'>[src] dispenses a meter!</span>")
+		else
+			var/obj/item/pipe/P = new(T, pipe_type=whatpipe, dir = usr.dir) //Now we make the pipes
+			if(iconrotation == 0 && Checkifbent(P.pipe_type) == 1) //Automatic rotation of dispensed pipes
+				P.dir = turn(usr.dir, 135)
+			else if(iconrotation == 0 && P.pipe_type in list(4, 7, 10, 17, 20, 32, 33, 34, 37)) //Some pipes dispense oppositely to what you'd expect, but we don't want to do anything if they selected a direction
```

Comment:
Though, making these values \#defines that have appropriate names would also be a good idea, so that future people aren't left guessing what these numbers mean